src/utils/translation/translate.ts

```diff
@@ -0,0 +1,55 @@
+// To use, remove lang attribute from index.html, import translate function, and call with translation key:
+// translate("~INTRO.HELLO")
+
+import enUS from "./lang/en-us.json";
+import es from "./lang/es.json";
+
+const languageFiles = [
+  {key: "en-US", contents: enUS},   // US English
+  {key: "es",    contents: es},     // Spanish
+];
+
+// returns baseLANG from baseLANG-REGION if REGION exists
+// this will, for example, convert en-US to en
+const getBaseLanguage = (langKey: any) => {
+  return langKey.split("-")[0];
+};
+
+// Get the HTML DOM lang property of the root element of the document
+const getPageLanguage = () => {
+  const pageLang = document.documentElement.lang;
+  return pageLang && (pageLang !== "unknown") ? pageLang : undefined;
+};
+
+// Get the first valid language specified by the browser
+const getFirstBrowserLanguage = () => {
+  const nav: any = window.navigator;
+  // userLanguage and browserLanguage are non standard but required for IE support
+  const languages = [...nav.languages, nav.language, nav.userLanguage, nav.browserLanguage];
+  return languages.find((browserLang) => browserLang);
+};
+
+const translations: any = {};
+
+languageFiles.forEach((langFile) => {
+  translations[langFile.key] = langFile.contents;
+  // accept full key with region code or just the language code
+  const bLang = getBaseLanguage(langFile.key);
+  if (bLang && !translations[bLang]) {
+    translations[bLang] = langFile.contents;
+  }
+});
+
+const lang = getPageLanguage() || getFirstBrowserLanguage();
+const baseLang = getBaseLanguage(lang || "");
+const defaultLang = lang && translations[lang]
+                    ? lang
+                    : baseLang && translations[baseLang]
+                      ? baseLang
+                      : "en";
+
+const translate = (key: string) => {
+  return translations[defaultLang][key] || `UNKNOWN KEY: ${key}`;
+};
```

Comment:
As mentioned in the comments, I think you should bring over the full `translate()` function. Variable replacement is required, and if CODAP/CFM found a use for the additional `lang` parameter I'd be willing to bet there was a good reason for it.

Reply:
I added these parameters and the additional functionality to the translate function.  No one needs to use them so I don't think they get in the way at all of a simple implementation.

Reply:
Thinking about this more, and following up on @scytacki's suggestion of thinking of this as the beginnings of a more widely used library within CC, I think we should change the API of the `translate()` function to be more extensible down the road. Something like the following:
```typescript
interface ITranslateOptions {
  lang?: string;
  count?: number;
  vars?: Record<string, string>;
}
export default function translate(str: string, options?: ITranslateOptions {
  ...
}
```

Reply:
Looks good.  I added these changes.